cfme/tests/infrastructure/test_provisioning_dialog.py

```diff
@@ -35,6 +34,13 @@
 ]
 
 
+def remove_vm(provider, vm_name):
+    if provider.mgmt.does_vm_exist(vm_name):
+        if provider.mgmt.is_vm_running(vm_name):
+            provider.mgmt.stop_vm(vm_name)
+        provider.mgmt.delete_vm(vm_name)
```

Comment:
I think that instead of creating two new instances of the same code you might like to put your changes into the cleanup_vm function. It was not deleting your VMs in your env?